File: content/code/Pet.cpp
#include <string>

class Pet {
 private:
  unsigned int hunger_{0};
  std::string name_{};

 public:
  explicit Pet(std::string name) : name_(name) {}
  void go_for_a_walk() { hunger_ += 1; }
  // ^-- how would you test this function?
  unsigned int hunger() const { return hunger_; }
};

File: content/code/Pet.R
Pet <- function(name) {
  structure(
    list(name = name, hunger = 0),
    class = "Pet"
  )
}

# How would you test this function?
take_for_a_walk <- function(pet) UseMethod("take_for_a_walk")
take_for_a_walk.Pet <- function(pet) {
  pet$hunger <- pet$hunger + 1
  pet
}
